lib/actions/api.js

```diff
@@ -30,25 +30,22 @@ export function routingQuery () {
     dispatch(routingRequest({ routingType, searchId }))
 
     // fetch a realtime route
-    fetch(constructRoutingQuery(otpState))
-      .then(getJsonAndCheckResponse)
-      .then(json => {
-        dispatch(routingResponse({ response: json, searchId }))
-      })
-      .catch(error => {
-        dispatch(routingError({ error, searchId }))
-      })
+    try {
+      const realtimeResponse = await fetch(constructRoutingQuery(otpState))
+      const realtimeJson = await getJsonAndCheckResponse(realtimeResponse)
+      await dispatch(routingResponse({ response: realtimeJson, searchId }))
```

Comment:
Are `getJsonAndCheckResponse` and `routingResponse` really async functions? Do we actually need to await them? I'm also curious if there was a reason this change was made. There is no comment associating the change, so it makes me think this is just coding style/for the heck of it?

Reply:
This change was more to make it easier to debug. It's easier to trace back errors with async/await as opposed to giant promise chains that catc errors that bubble up from dispatches unrelated to fetching. I can remove this if you like.

And yes, all of these things can be awaited. See [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) and [redux-thunk dispatch chaining](https://github.com/reduxjs/redux-thunk#composition).